src/controllers/game-controller.js

```diff
@@ -0,0 +1,73 @@
+import { showGameOverMessage, showGameStartMessage, showNumberInputMessage, showRestartMessage, showUpAndDownStatus } from '../views/game-view.js';
+import { MAX_COUNT, MAX_RANDOM_NUMBER, MIN_RANDOM_NUMBER } from '../constants.js';
+import { showErrorMessage } from '../views/error-message.js';
+import { getRandomNumber, getUpAndDownStatus, isValidUserInput, updatePrevInputAndCount } from '../models/game-model.js';
+
+export async function askRestart() {
+    while (true) {
+        const playAgainInput = await showRestartMessage()
+
+        if (playAgainInput === 'no') {
+            showGameOverMessage()
+            break;
+        }
+        if (playAgainInput === 'yes') {
+            playGame();
+            break;
+        }
+    }
+}
+
+async function playGame() {
+    showGameStartMessage()
+    const randomNumber = getRandomNumber(MIN_RANDOM_NUMBER, MAX_RANDOM_NUMBER);
+
+    console.log(randomNumber);
+
+    const prevResult = {
+        prevInput: [],
+        count: 0,
+    };
+
+    while (true) {
+        const userInput = await showNumberInputMessage()
+
+        if (isValidUserInput({ userInput })) {
```

Comment:
요렇게 매개변수를 넘겨주는게 일반적일 것 같아요~
```suggestion
        if (isValidUserInput(userInput)) {
```